source/reference/compatibility.txt

```diff
@@ -128,6 +128,32 @@ Write Preference Behavior
 legacy :binary:`mongo` shell. To disable retryable writes, use 
 :option:`--retryWrites=false <mongosh --retryWrites>`.
 
+
+ObjectId Methods and Attributes
+-------------------------------
+
+The ObjectId() methods work differently in ``mongosh`` than in the
+legacy ``mongo`` shell.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Method or Attribute
+     - ``mongo`` Behavior
+     - ``mongosh`` Behavior
+
+   * - ``ObjectId.str``
+     - Hexadecimal string
+     - Undefined
+
+   * - ``ObjectId.valueOf()``
+     - Returns ``ObjectId.str``
+     - Returns ``ObjectId.self``
+
+   * - ``ObjectId.toString()``
+     - Returns an ObjectId("<hexadecimal>") string
```

Comment:
I think we should monospace ObjectId("<hexadecimal>"). I might also consider rephrasing this to be something along the lines of:

"Returns a string representation of the ObjectId in the format ``ObjectId("<hexadecimal>")``."